### ISSUE TITLE
Why does the connection keeps closing automatically after a few hours?

### ISSUE THREAD

**Issue by Manjit2003** [adiwajshing/Baileys#169]
@adiwajshing I have created a connection on my AWS server but after about every 7-8 hours the connection closed automatically how ever my WhatsApp is still showing "WhatsApp web is Active"

**Comment by adiwajshing** [comment]
Can't help you without more details like code or a log

**Comment by Manjit2003** [comment]
```
import { WAConnection } from '@adiwajshing/baileys'

async function connectToWhatsApp () {
    const conn = new WAConnection() 
    
    await conn.connect ()
    console.log ("oh hello " + conn.user.name + " (" + conn.user.id + ")")
    // every chat object has a list of most recent messages
    console.log ("you have " + conn.chats.all().length + " chats")

    const unread = await conn.loadAllUnreadMessages ()
    console.log ("you have " + unread.length + " unread messages")
}
// run in main file
connectToWhatsApp ()
.catch (err => console.log("unexpected error: " + err) ) // catch any errors
while(true){setTimeout(function(){console.log('runnig')}, 100000)}
```

**Comment by Manjit2003** [comment]
but it gets disconnected automatically....

**Comment by bhavya32** [comment]
You should use - 
```
conn.autoReconnect = ReconnectMode.onAllErrors;
conn.connectOptions.maxRetries = 10000;
```

**Comment by bhavya32** [comment]
@adiwajshing I am also facing the same issue. If I turn off my phone's internet, start baileys, it loops for reconnection, but if I turn on my mobile data, it still loops between ```sending init queries```  and retrying.

**Comment by TheBlueKingLP** [comment]
I’m having the same issue

**Comment by bhavya32** [comment]
@adiwajshing Please I am not asking you to solve it but at least acknowledge if you are aware or is it our side only?

**Comment by adiwajshing** [comment]
@bhavya32 I'm maintaining this repo in my free time & for free. Sometimes, emails slip & if you expect more robust service from me, you can hire me for a fee.

However, until then, I ask you to be a little patient & respectful. Moreover, I would ask you to attempt to solve the issue on your own, if you can. The community would be very grateful as well.

**Comment by adiwajshing** [comment]
Speaking to the original issue, I cannot reproduce it. It works fine when I test with the phone initially off.

So:
1. Try increasing `maxIdleTimeMs` to about 30 seconds.
2. Also, no log has been provided. Please do provide one.

**Comment by bhavya32** [comment]
@adiwajshing calm down bro. I think you took my request the wrong way. I was just confirming if you were also facing the error. I did not asked you to solve it. I will contribute for sure if I solved it. For then - the log goes this way - 
```
[Baileys][10/12/2020, 1:53:42 PM] loading authentication credentials from ./tokens/client-token.json
[Baileys][10/12/2020, 1:53:42 PM] connected to WhatsApp Web server, authenticating via takeover
[Baileys][10/12/2020, 1:54:42 PM] cancelling message wait: s1
[Baileys][10/12/2020, 1:54:42 PM] connect attempt 1 failed: BaileysError: timed out, retrying...
[Baileys][10/12/2020, 1:54:44 PM] connected to WhatsApp Web server, authenticating via takeover
[Baileys][10/12/2020, 1:55:44 PM] cancelling message wait: s1
[Baileys][10/12/2020, 1:55:44 PM] connect attempt 2 failed: BaileysError: timed out
```
It continues forever until ```maxretries``` is reached. I did not got it on low wait times like 5 minutes but surely when I turned off wifi for whole night. Time between s1 and s2 is not affected by ```maxidletimems```. It always takes 1 minute.

**Comment by Manjit2003** [comment]
@bhavya32 got a solid fix...

just add - 

```
conn.on('close', async (reason, isReconnecting) => {
    
    console.log("Disconnected because " +reason.reason)
    
    if(reason.reason === 'lost'){
        conn.close()
        conn.loadAuthInfo(conn.base64EncodedAuthInfo ())
        conn.connect()
        return {}
    }
    if(reason.reason != 'intentional'){
        console.log('Unable to reconnect')
    }
})

```
It just auto-reconnects by using the auth credentials... But be sure to set  **`conn.autoReconnect = 0`** while starting the connection...

Hope it helps as in my case it worked like a charm.

**Comment by bhavya32** [comment]
@Manjit2003 Thanks for sharing but I already tried implementing that a day before. I get a temporary ban from whatsapp doing so (not on mobile but accessing one of the IP address of whatsapp)and this results in closing of session.  I do not know maybe it was some other issue but definitely gonna try once more.

**Comment by bhavya32** [comment]
@Manjit2003 ```conn.connectOptions.alwaysUseTakeover = true``` in the latest commit solves this. @adiwajshing ,the issue is closed with commit - 16b138c . Thank you.
